Minor edits for the supply chain compromise catalogue

joshuagl opened the pull request:
Most notably this expands the definition of negligence and adds a definition of attack chaining, both suggested in #407. I've also added some notes on mitigating dev tooling attacks.

lumjjb commented on supply-chain-security/compromises/compromise-definitions.md:
This seems more of a technique rather than a type of compromise. It doesn't quite "typecheck" with the above headers. I think this is valuable information, how about putting it as an addon to the paragraph of types of supply chains on top - kind of like here are some types of supply chain compromises, sometimes, a breach may be attributed to multiple lapses, etc.

TheFoxAtWork replied:
@lumjjb It may be beneficial to list it here, even though this is technique rather than true compromise type, multiple compromises occurring together for the purposes of this catalog showcases multiple weaknesses at a top level.  Also, some of the attacks in the catalog are unclear in the reporting which compromise was the initial, or true start of the compromises.  
Perhaps not including it in a compromise, but as an annotation below the definitions and listing in the compromise type table at the top level to improve readability?

lumjjb replied:
Yea agreed, suggested top, but bottom is fine too, has the same effect.

joshuagl replied:
Thank you both for the feedback. I'm not quite sure what to do here. So far as I know, we can only anchor on headers so in order to include attack chaining as a clickable link in the compromise type list we need a header to link to.

We could add a heading level 1 above attack chaining? Or change the attack chaining header to include technique in the title somewhere? I've pushed a commit to this PR which follows the second option, adding "technique" to the header, and cleans up the description a bit.

lumjjb commented on supply-chain-security/compromises/compromise-definitions.md:
I think it would be nice to have "Occurs due to a lack of adherence to best practices" to be at the start of it - to describe what negligence refers to in this context, and typosquatting  attacks are an example of such negligence.

joshuagl replied:
Thanks for the suggestion. I've rephrased this a bit, please let me know what you think.

lumjjb commented on supply-chain-security/compromises/compromise-definitions.md:
Caps Use

joshuagl commented:
I've pushed some additional patches to try and address feedback here. I'm not sure if I've completely addressed the comment on technique vs. type on attack chaining. I'd welcome additional feedback on that.

TheFoxAtWork commented:
@joshuagl this looks great!!  @lumjjb can you take a look?

SantiagoTorres commented:
LGTM! I was going to propose the second option regarding [this](https://github.com/cncf/sig-security/pull/408#discussion_r460820020) discussion too. Thank you, @joshuagl, @lumjjb, @TheFoxAtWork !
